src/run/run.rs

```diff
@@ -51,6 +51,16 @@ impl PartialEq for ComparisonGenerators {
     }
 }
 
+/// Error type for an invalid comparison name
+#[derive(PartialEq, Debug)]
+pub enum ComparisonError {
```

Comment:
You probably want to use quick_error! to define this. Look at the xml_util.rs code to see how to do that. That's also where you want to define the ComparisonResult type. (I think xml_util defines its own Result type too, so that's a good reference).